Optimizer/Minuit2/MinuitResult.cpp

```diff
@@ -2,275 +2,44 @@
-  }
-  InitialLH = -1;
-  FinalLH = minState.Fval();
-  Edm = minState.Edm();
-  IsValid = min.IsValid();
-  CovPosDef = min.HasPosDefCovar();
-  HasValidParameters = min.HasValidParameters();
-  HasValidCov = min.HasValidCovariance();
-  HasAccCov = min.HasAccurateCovar();
-  HasReachedCallLimit = min.HasReachedCallLimit();
-  EdmAboveMax = min.IsAboveMaxEdm();
-  HesseFailed = min.HesseFailed();
-  ErrorDef = min.Up();
-  NFcn = min.NFcn();
-
-  return;
-}
-
-void MinuitResult::genOutput(std::ostream &out, std::string opt) {
-  bool printParam = 1, printCorrMatrix = 1, printCovMatrix = 1;
-  if (opt == "P") { // print only parameters
-    printCorrMatrix = 0;
-    printCovMatrix = 0;
-  }
-  out << std::endl;
-  out << "--------------MINUIT2 FIT RESULT----------------" << std::endl;
+  os << "--------------MINUIT2 FIT INFOS----------------\n";
   if (!IsValid)
-    out << "    *** MINIMUM NOT VALID! ***" << std::endl;
-  out << std::setprecision(10);
-  out << "Initial Likelihood: " << InitialLH << std::endl;
-  out << "Final Likelihood: " << FinalLH << std::endl;
-  if (TrueLH)
-    out << "True Likelihood: " << TrueLH << std::endl;
-
-  out << "Estimated distance to minimumn: " << Edm << std::endl;
+    os << "    *** MINIMUM NOT VALID! ***\n";
+  os << "Estimated distance to minimum: " << Edm << "\n";
   if (EdmAboveMax)
-    out << "    *** EDM IS ABOVE MAXIMUM! ***" << std::endl;
-  out << "Error definition: " << ErrorDef << std::endl;
-  out << "Number of calls: " << NFcn << std::endl;
+    os << "    *** EDM IS ABOVE MAXIMUM! ***\n";
+  os << "Error definition: " << ErrorDef << "\n";
+  os << "Number of calls: " << NFcn << "\n";
   if (HasReachedCallLimit)
-    out << "    *** LIMIT OF MAX CALLS REACHED! ***" << std::endl;
-  out << "CPU Time : " << Time / 60 << "min" << std::endl;
-  out << std::setprecision(5) << std::endl;
-
+    os << "    *** LIMIT OF MAX CALLS REACHED! ***\n";
   if (!HasValidParameters)
-    out << "    *** NO VALID SET OF PARAMETERS! ***" << std::endl;
-  if (printParam) {
-    out << "PARAMETERS:" << std::endl;
-    TableFormater *tableResult = new TableFormater(&out);
-    printFitParameters(tableResult);
-  }
-
+    os << "    *** NO VALID SET OF PARAMETERS! ***\n";
   if (!HasValidCov)
-    out << "    *** COVARIANCE MATRIX NOT VALID! ***" << std::endl;
+    os << "    *** COVARIANCE MATRIX NOT VALID! ***\n";
   if (!HasAccCov)
-    out << "    *** COVARIANCE MATRIX NOT ACCURATE! ***" << std::endl;
+    os << "    *** COVARIANCE MATRIX NOT ACCURATE! ***\n";
   if (!CovPosDef)
-    out << "    *** COVARIANCE MATRIX NOT POSITIVE DEFINITE! ***" << std::endl;
+    os << "    *** COVARIANCE MATRIX NOT POSITIVE DEFINITE! ***\n";
   if (HesseFailed)
-    out << "    *** HESSE FAILED! ***" << std::endl;
-  if (HasValidCov) {
-    if (printCovMatrix) {
-      out << "COVARIANCE MATRIX:" << std::endl;
-      TableFormater *tableCov = new TableFormater(&out);
```

Comment:
Is the covariance not printed?

Reply:
The covariance matrix is now part of the general FitResult. So it will be printed if it is non empty